### PR TITLE
feat: base of TextField component

### PR DESCRIPTION
**Pull request by Michael-Liendo**

### CODE REVIEW

File: crates/web/src/components/text_field.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+use leptos::{component, create_memo, view, IntoView, MaybeProp, SignalGet};
+
+#[component]
+pub fn TextField(
+    #[prop(optional, into)] name: String,
+    #[prop(optional, into)] id: String,
+    #[prop(optional, into)] placeholder: String,
+    #[prop(optional, into)] value: String,
+    #[prop(optional, into)] label: String,
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:11]: I think `leptos::TextProp` its more convenient here

File: crates/web/src/components/text_field.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    #[prop(optional, into, default = "text".to_string())] r#type: String,
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:12]: Can we have a `TextFieldVariant` enum for this?

File: crates/web/src/components/text_field.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    #[prop(optional, into)] disabled: MaybeProp<bool>,
+    #[prop(optional, into)] full_width: MaybeProp<bool>,
+) -> impl IntoView {
+    println!("label {}", label);
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:16]: I recommend `leptos::logging::log` macro here, its compatible with browser console

**Review by EstebanBorai — COMMENTED** [review]
Good progress @Michael-Liendo! Im leaving early comments, keep it up!

**Comment by Michael-Liendo** [comment]
> Good progress @Michael-Liendo! Im leaving early comments, keep it up!

thanks you!

File: crates/web/src/components/text_field.rs (hunk 2 above)
Comment (Michael-Liendo) [reply crates/web/src/components/text_field.rs:12]: Sure!

File: crates/web/src/components/text_field.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+#[component]
+pub fn TextField(
+    #[prop(optional, into)] name: TextProp,
+    #[prop(optional, into)] id: TextProp,
+    #[prop(optional, into)] placeholder: TextProp,
+    #[prop(optional, into)] value: TextProp,
+    #[prop(optional, into)] label: TextProp,
+    #[prop(optional, into)] variant: MaybeProp<TextFieldVariant>,
+    #[prop(optional, into, default = "text".to_string())] r#type: String,
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:19]: Can we use an enum for this as well please?

File: crates/web/src/components/text_field.rs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                classes.insert("!cursor-not-allowed");
+            }
+        }
+
+        classes.into_iter().collect::<Vec<&str>>().join(" ")
+    });
+    view! {
+            <div>
+            <label class="block mb-2 text-sm font-medium text-purple-500" for=id.clone()>{label}</label>
+            <input type=format!("{:?}", r#type).to_lowercase() name=name value=value id=id placeholder=placeholder class=class_names  disabled=disabled />
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:73]: Can we have a `Display` trait implementation for `TextFieldType` so we have a explicit value for each type?

I also suggest narrowing down a bit the variants to just:

- Text
- Email
- Password

```rust
            <input type=format!("{:?}", r#type).to_lowercase()
```

Comment (Michael-Liendo) [reply crates/web/src/components/text_field.rs:73]: Okey!! 🚀

File: crates/web/tests/components/text_field.rs
Comment (EstebanBorai) [inline crates/web/tests/components/text_field.rs:1]: Can we have tests for the `Type` enum we just introduced?

**Review by EstebanBorai — COMMENTED** [review]
Great job @Michael-Liendo!

Just one minor chore and I think we can merge this!

File: crates/web/tests/components/text_field.rs
Comment (Michael-Liendo) [reply crates/web/tests/components/text_field.rs:1]: Yeah, sure!!

**Review by EstebanBorai — APPROVED** [review]
Great!

**Pull request merged**